pkg/applicationserver/applicationserver.go

```diff
@@ -1352,6 +1414,28 @@ func (as *ApplicationServer) handleDownlinkNack(ctx context.Context, ids *ttnpb.
 			}
 
 			items := []*ttnpb.ApplicationDownlink{msg}
+
+			if msg.Confirmed {
+				if msg.ConfirmedRetry == nil {
+					msg.ConfirmedRetry = &ttnpb.ApplicationDownlink_ConfirmedRetry{
+						Attempt: 0,
+					}
+				}
+				msg.ConfirmedRetry.Attempt++
+
+				maxRetries := uint32(as.config.Downlinks.ConfirmationConfig.DefaultRetryAttempts)
+				if msg.ConfirmedRetry.MaxAttempts.GetValue() > 0 {
+					maxRetries = msg.ConfirmedRetry.MaxAttempts.GetValue()
+				}
+				if maxRetries != 0 && msg.ConfirmedRetry.Attempt > maxRetries {
```

Comment:
In which situations would this be zero, if we make the AS configuration not accept zeroes for the default or (validation) max ?